src/pgp.rs

```diff
@@ -240,19 +240,14 @@ fn select_pk_for_encryption(key: &SignedPublicKey) -> Option<SignedPublicKeyOrSu
 /// and signs it using `private_key_for_signing`.
 pub fn pk_encrypt(
     plain: &[u8],
-    public_keys_for_encryption: &Keyring,
+    public_keys_for_encryption: &Keyring<SignedPublicKey>,
     private_key_for_signing: Option<&Key>,
 ) -> Result<String> {
     let lit_msg = Message::new_literal_bytes("", plain);
     let pkeys: Vec<SignedPublicKeyOrSubkey> = public_keys_for_encryption
         .keys()
         .iter()
-        .filter_map(|key| {
-            key.as_ref()
-                .try_into()
-                .ok()
-                .and_then(select_pk_for_encryption)
-        })
+        .filter_map(|key| select_pk_for_encryption(key))
```

Comment:
The `.keys()` method now returns `&T` instead of `Cow<T>` because that's what all the users needed.  This strips the layer of `.as_ref()` here (and in similar calls below).

The rest of this wrangling falls away because the keyring now directly contains the rpgp `SignedPrivateKey` types so there's less conversions in these calls (similar calls below but without the `select_pk_for_encryption()` bit which simplifies them even more).